### ISSUE TITLE
Override commit.gpgsign when creating patch-file

### ISSUE THREAD

**Issue by FibreFoX** [ds300/patch-package#456]
I have a local setup, where my **global** git-settings contains the following setting:
```ini
[commit]
  gpgsign = true
```

This creates a problem while creating a patch-file, which fails because there is no git-key configured (I manage these on local repository level ... yes I am weird, but this forces me to sign my commits properly).

I've created a patch-file for this (haha), which explicitly overrides that setting:
```patch
diff --git a/node_modules/patch-package/dist/makePatch.js b/node_modules/patch-package/dist/makePatch.js
index a76e39c..86dce12 100644
--- a/node_modules/patch-package/dist/makePatch.js
+++ b/node_modules/patch-package/dist/makePatch.js
@@ -117,6 +117,8 @@ function makePatch({ packagePathSpecifier, appPath, packageManager, includePaths
         git("init");
         git("config", "--local", "user.name", "patch-package");
         git("config", "--local", "user.email", "patch@pack.age");
+        // do not sign this temporary commit
+        git("config", "--local", "commit.gpgsign", "false");
         // remove ignored files first
         filterFiles_1.removeIgnoredFiles(tmpRepoPackagePath, includePaths, excludePaths);
         git("add", "-f", packageDetails.path);
```

Would you prefer a pull-request for this one?

**Comment by simon-abbott** [comment]
@ds300 This issue has been open for nearly three years with a really simple fix that even [has a PR](https://github.com/ds300/patch-package/pull/508). I run into this issue every single time I try to use `patch-package`, are there any plans to fix this?

**Comment by FibreFoX** [comment]
@simon-abbott at least patch-package can patch itself to "fix" this ;) at least thats what I am doing in my projects until this got merged

**Comment by simon-abbott** [comment]
True! I should do that. But it's annoying to have to remember to add that to every new project.

**Comment by simon-abbott** [comment]
Oh wait I just realized it's actually been known for almost 4.5 years, again with a proposed fix: https://github.com/ds300/patch-package/issues/353. It's been so long I forgot I even left a comment on the other issue!

**Comment by FibreFoX** [comment]
Oh, interesting, did not see that patch back in the days ... so I doubt this fix will ever get merged, sadly .... so self-patching it is, such irony
